aea/test_tools/test_cases.py

```diff
@@ -947,6 +990,47 @@ def teardown_class(cls) -> None:
         super(AEATestCaseMany, cls).teardown_class()
 
 
+class AEATestCaseManyFlaky(AEATestCaseMany):
+    """
+    Test case for many AEA projects which are flaky.
+
+    Use for flaky tests with the flaky decorator.
+    """
+
+    failure_count: int = 0
+
+    @classmethod
+    def setup_class(cls) -> None:
+        """Set up the test class."""
+        method_list = [
+            func
+            for func in dir(cls)
+            if callable(getattr(cls, func))
+            and not func.startswith("__")
+            and func.startswith("test_")
+        ]
+        if len(method_list) > 1:
+            raise ValueError(f"{cls.__name__} can only contain one test method!")
+        super(AEATestCaseManyFlaky, cls).setup_class()
+        cls.failure_count = 0
+
+    def setup(self) -> None:
+        """Set up test case instance."""
+        if self.failure_count == 0:
+            return
+        self.teardown_class()
+        self.setup_class()
+
+    def teardown(self) -> None:
+        """Tear down test case instance."""
+        self.failure_count += 1
+
+    @classmethod
+    def teardown_class(cls) -> None:
+        """Teardown the test class."""
+        super(AEATestCaseManyFlaky, cls).teardown_class()
+
+
```

Comment:
this looks like redundant code. It could be solved by using a Mixin approach?

Reply:
I'm not that upset though. We have not violated the rule of three ;-)
https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming)

Reply:
I was thinking to use a mixin but then there can be MRO issues when doing the super calls?

Reply:
you can call specific super-methods by prepending the name of the class:

```
MixIn.setup_class(cls)
AEATestCaseMany.setup_class(cls)
```

MRO issues could happen only at definition-time of the class, not at runtime.

Reply:
Yes but this is one more thing the users of this class need to get right.

Reply:
The class that inherits from `AEATestCaseManyFlaky`, in case needs to override `setup_class`, can use 
```
super(AEATestCaseManyFlaky, cls).setup_class()
```

Reply:
anyway, as I said, this is good enough.